File: app-mvp/src/main/java/xuk/imaginary/gui/gank/meizhi/MeizhiContract.java
package xuk.imaginary.gui.gank.meizhi;

import java.util.List;

import xuk.imaginary.BasePresenter;
import xuk.imaginary.BaseView;
import xuk.imaginary.data.BaseBean;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface MeizhiContract {

    interface View extends BaseView<Presenter> {

        void addList(boolean isRefresh, List<BaseBean> meiZhis);
    }

    interface Presenter extends BasePresenter {
        void getList(boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/xuk/imaginary/gui/gank/meizhi/MeizhiContract.kt
package xuk.imaginary.gui.gank.meizhi

import xuk.imaginary.BasePresenter
import xuk.imaginary.BaseView
import xuk.imaginary.data.BaseBean

/**
 * This specifies the contract between the view and the presenter.
 */
interface MeizhiContract {

  interface View : BaseView<Presenter> {

    fun addList(isRefresh: Boolean, meiZhis: List<BaseBean>)
  }

  interface Presenter : BasePresenter {
    fun getList(isRefresh: Boolean)
  }
}
